x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engine/engine_schema.tsx

```diff
@@ -143,12 +144,23 @@ const SchemaFieldDetails: React.FC<{ schemaField: SchemaField }> = ({ schemaFiel
 
 export const EngineSchema: React.FC = () => {
   const { engineName } = useValues(EngineIndicesLogic);
+  const [onlyShowConflicts, setOnlyShowConflicts] = useState<boolean>(false);
   const { isLoadingEngineSchema, schemaFields } = useValues(EngineViewLogic);
   const { fetchEngineSchema } = useActions(EngineViewLogic);
   const [itemIdToExpandedRowMap, setItemIdToExpandedRowMap] = useState<Record<string, JSX.Element>>(
     {}
   );
 
+  const toggleOnlyShowConflicts = useCallback(() => {
```

Comment:
suggestion: I think it's a little cleaner to split these two up. You probably want to call `setItemIdToExpandedRowMap` whenever `onlyShowConflicts` changes. Now you're doing that implicitly, inside the callback, but it's cleaner (and easier to follow) if you do that in a separate useEffect that gets triggered by the state changing:
```
useEffect(() => {
    setItemIdToExpandedRowMap({});
  }, [onlyShowConflicts]
};
```
Then you can lose the `useCallBack` and just directly call `setOnlyShowConflicts(!onlyShowConflicts)` in the toggle below.

Reply:
✅ https://github.com/elastic/kibana/pull/154149/commits/254c9ae52fbf0c8e931896e8d786a7b7a3362aee

Reply:
i personally find this callback approach easier to trace but your suggested `useEffect` approach also causes a double render.

1. `setOnlyShowConflicts` updates state which causes a render
2. `onlyShowConflicts` has changed in the `useEffect` dependencies so the effect is run
3. `setItemIdToExpandedRowMap` is called updating state causing the second render

here are flame graphs for each option

|  | click event duration | 📊 |
| - | - | -|
| existing `useCallback` | 35.5ms | https://share.firefox.dev/40AXlA5 |
| proposed `useEffect` | 72.2ms | https://share.firefox.dev/3K9expn |

Reply:
Yes, that's the downside of separating all of this out. I tend to prefer the traceability of separating these dependencies out (I've seen too many cases where stacking effects leads to weird bugs down the road), but if you find this more readable it shouldn't be an issue. Approved :)